tensorflow_probability/python/mcmc/transformed_kernel.py

```diff
@@ -221,6 +246,33 @@ def make_likelihood(true_variances):
       axis=0)
   ```
 
+  #### Detailed API Contract
+
+  `TransformedTransitionKernel` operates on a `target_log_prob_fn` field, which
+  is expected to be an attribute of some `inner_kernel` (which may or may not
+  be the most immediate). If the provided `inner_kernel` doesn't have a
+  `target_log_prob_fn`, `TransformedTransitionKernel` will iteratively look
+  through deeper layers of kernels to find one that does, stopping at the first
```

Comment:
When you say "deeper layers of kernels", be explicit that it searches in the `inner_kernel` field, if present, at each level.  Also find a way to make that sentence not awkward :)

Reply:
why don't we copy this docstring over to a doc to collaboratively edit; i am not sure github's comment interface is going to be conducive to this task

Reply:
Good idea! I'll make a doc for this

Reply:
i made some (substantial) edits to the doc -- PTAL!